tests/ui/landing_page_test.py

```diff
@@ -42,75 +45,231 @@ def testLandingPageLayout(self):
+
+    # Get the initial invite code.
+    initial_invite_code = details_dialog.find_element(
+        *LandingPage.USER_INVITE_CODE_TEXT).get_attribute('value')
+
+    # Click new invite code on that user.
+    get_invite_code_button = details_dialog.find_element(
+        *LandingPage.USER_ROTATE_KEYS_BUTTON)
+    get_invite_code_button.click()
+
+    # Wait for post to finish, can take a while.
+    WebDriverWait(self.driver, self.DEFAULT_TIMEOUT).until(
+        EC.invisibility_of_element_located(((
+            LandingPage.USER_DETAILS_SPINNER))))
+
+    # Renavigate to the landing page to ensure the backend changed vs just a
+    # UI update if specified.
+    self.driver.get(self.args.server_url + flask.url_for('landing'))
+    test_user_item = self.findTestUserOnLangingPage()
+
+    # Set the container element to find the disable/enable button and text on.
+    # Will be either the item within the listbox or the overall details dialog.
+    details_dialog = self.getDetailsDialogForItem(test_user_item)
+
+    # Check the invite code text changed.
+    final_invite_code = details_dialog.find_element(
+        *LandingPage.USER_INVITE_CODE_TEXT).get_attribute('value')
+    self.assertNotEquals(initial_invite_code, final_invite_code)
 
   def testAddServerFromLandingPage(self):
     """Test that adding a server shows up on the server listing."""
-    self.assertTestServerPresenceOnLandingPage(False)
+    self.assertTestServerPresenceOnPage(False)
 
-    self._addTestServerFromLandingPage()
+    self.addTestServerFromLandingPage()
 
-    self.assertTestServerPresenceOnLandingPage(True)
+    self.assertTestServerPresenceOnPage(True)
 
   def testDeleteServer(self):
     """Test that deleting a server removes that server."""
-    self._addTestServerFromLandingPage()
+    self.addTestServerFromLandingPage()
 
-    self.assertTestServerPresenceOnLandingPage(True)
+    self.assertTestServerPresenceOnPage(True)
 
     self.removeTestServer()
 
-    self.assertTestServerPresenceOnLandingPage(False)
+    self.assertTestServerPresenceOnPage(False)
 
   def testDownloadChromePolicyFromLandingPage(self):
     """Test that the chrome policy download link is present and wired up."""
     self.driver.get(self.args.server_url + flask.url_for('landing'))
     self.assertChromePolicyDownloadLinkIsConnected()
 
-  def _addTestUserFromLandingPage(self):
-    """Manually add a test user using the landing page."""
-    # Navigate to add user and go to manual tab.
-    self.driver.get(self.args.server_url + flask.url_for('landing'))
-    landing_page = LandingPage(self.driver)
-    add_user_button = landing_page.GetElement(LandingPage.ADD_USER_BUTTON)
-    add_user_button.click()
-    add_manually_tab = WebDriverWait(self.driver, self.DEFAULT_TIMEOUT).until(
-        EC.visibility_of_element_located(((LandingPage.ADD_MANUALLY_TAB))))
-    add_manually_tab.click()
+  def enableDisableTestHelper(self, should_use_listbox, should_refresh_page):
```

Comment:
nit: Look at what this method is doing, maybe the naming here should be "disableAndThenEnableUser()".